src/centralserver/admin-service/src/test/java/org/niis/xroad/centralserver/restapi/openapi/InitializationApiControllerTest.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    @Test
+    public void initCentralServer() throws Exception {
+        when(signerProxyFacade.getTokens()).thenReturn(Collections.singletonList(testSWToken));
+
+        ResponseEntity<Void> response = initializationApiController.initCentralServer(okConf);
+        assertNotNull(response);
+        assertEquals(200, response.getStatusCodeValue());
+
+        List<SystemParameter> storedCentralServerAddresses = systemParameterRepository
+                .findSystemParametersByKey(CENTRAL_SERVER_ADDRESS);
+        assertEquals(1, storedCentralServerAddresses.size());
+        assertEquals(okConf.getCentralServerAddress(), storedCentralServerAddresses.iterator().next().getValue());
+        ResponseEntity<InitializationStatus> statusResponseEntity =
+                initializationApiController.getInitializationStatus();
+        assertNotNull(statusResponseEntity.getBody());
+        assertEquals(
+                TokenInitStatus.INITIALIZED,
+                statusResponseEntity.getBody().getSoftwareTokenInitStatus()
+        );
+        assertEquals("TEST", statusResponseEntity.getBody().getInstanceIdentifier());
+        assertEquals("xroad.example.org", statusResponseEntity.getBody().getCentralServerAddress());
+    }
+
+    @Test
+    public void initCentralServerMissingParams() {
+        InitialServerConf testConf = new InitialServerConf();
+        testConf.instanceIdentifier("TEST").centralServerAddress("xroad.example.org").softwareTokenPin(null)
+                .ignoreWarnings(false);
+        ConstraintViolationException constraintViolationException = assertThrows(ConstraintViolationException.class,
+                () -> initializationApiController.initCentralServer(testConf));
+        assertEquals(1, constraintViolationException.getConstraintViolations().size());
+
+    }
+
+    @Test
+    public void initCentralServerAlreadyInitialized() throws Exception {
+        SystemParameter validCentralServerAddressParameter = new SystemParameter();
+        validCentralServerAddressParameter.setKey(CENTRAL_SERVER_ADDRESS);
+        validCentralServerAddressParameter.setValue("123.123.123.123");
+
+        SystemParameter validInstanceIdentifierParameter = new SystemParameter();
+        validInstanceIdentifierParameter.setKey(CENTRAL_SERVER_ADDRESS);
+        validInstanceIdentifierParameter.setValue("VALID_INSTANCE");
+
+        when(signerProxyFacade.getTokens()).thenReturn(Collections.singletonList(testSWToken));
+
+        when(systemParameterRepository.findSystemParametersByKey(
+                CENTRAL_SERVER_ADDRESS)
+        ).thenReturn(List.of(validCentralServerAddressParameter));
+
+        when(systemParameterRepository.findSystemParametersByKey(
+                INSTANCE_IDENTIFIER)
+        ).thenReturn(List.of(validInstanceIdentifierParameter));
+        assertThrows(ConflictException.class, () -> initializationApiController.initCentralServer(okConf));
```

Comment:
Could be written without mocking by initializing the central server twice, and checking that the second initialization fails.

Reply:
done